drivers/mcp2515/candev_mcp2515.c

```diff
@@ -0,0 +1,582 @@
+            if (value_len < sizeof(candev->bittiming)) {
+                res = -EOVERFLOW;
+            }
+            else {
+                memcpy(&candev->bittiming, value, sizeof(candev->bittiming));
+                res = _init(candev);
+                if (res == 0) {
+                    res = sizeof(candev->bittiming);
+                }
+
+            }
+            break;
+        case CANOPT_STATE:
+            if (value_len < sizeof(uint8_t)) {
+                res = -EOVERFLOW;
+            }
+            else {
+                switch (*((canopt_state_t *)value)) {
+                    case CANOPT_STATE_LISTEN_ONLY:
+                        res = mcp2515_set_mode(dev, MODE_LISTEN_ONLY);
+                        break;
+                    case CANOPT_STATE_OFF:
+                    case CANOPT_STATE_SLEEP:
+                        res = mcp2515_set_mode(dev, MODE_SLEEP);
+                        break;
+                    case CANOPT_STATE_ON:
+                        res = mcp2515_set_mode(dev, MODE_NORMAL);
+                        break;
+                    default:
+                        res = -ENOTSUP;
+                        break;
+                }
+            }
+            break;
+        default:
+            res = -ENOTSUP;
+    }
+
+    return res;
+}
+
+static int _get(candev_t *candev, canopt_t opt, void *value, size_t max_len)
+{
+    candev_mcp2515_t *dev = (candev_mcp2515_t *)candev;
+    int res = 0;
+
+    DEBUG("Inside mcp2515 get opt=%d\n", opt);
+    switch (opt) {
+        case CANOPT_BITTIMING:
+            if (max_len < sizeof(candev->bittiming)) {
+                res = -EOVERFLOW;
+            }
+            else {
+                memcpy(value, &candev->bittiming, sizeof(candev->bittiming));
+                res = sizeof(candev->bittiming);
+            }
+            break;
+        case CANOPT_RX_FILTERS:  /**< rx filters */
+            if (max_len % sizeof(struct can_filter) != 0) {
+                res = -EOVERFLOW;
+            }
+            else {
+                res = 1;
+            }
+            break;
+        case CANOPT_BITTIMING_CONST:
+            if (max_len < sizeof(bittiming_const)) {
+                res = -EOVERFLOW;
+            }
+            else {
+                memcpy(value, &bittiming_const, sizeof(bittiming_const));
+                res = sizeof(bittiming_const);
+            }
+            break;
+        case CANOPT_CLOCK:
+            if (max_len < sizeof(uint32_t)) {
+                res = -EOVERFLOW;
+            }
+            else {
+                *((uint32_t *)value) = (uint32_t)(dev->conf->clk / 2);
+                res = sizeof(uint32_t);
+            }
+            break;
+        default:
+            res = -ENOTSUP;
+            break;
+    }
+
+    return res;
+}
+
+static int _set_filter(candev_t *dev, const struct can_filter *filter)
+{
+    DEBUG("inside _set_filter of MCP2515\n");
+    int filter_added = -1;
```

Comment:
Is this the convention of doing it in RIOT? Because as far as I'm aware you don't really gain any advantage, but you will need to include this extra header (stdbool.h).

(I changed it to bool for now)